src/ocrd_anybaseocr/ocrd-tool.json

```
@@ -0,0 +1,136 @@
+          "description": "ruler detection and suppression: minimum area of bbox (as ratio of total image pixels)"
+        },
+        "rulerWidthMax": {
+          "type": "number",
+          "format": "float",
+          "default": 0.95,
+          "description": "ruler detection and suppression: maximum width of bbox (as ratio of total image width)"
+        },
+        "columnAreaMin": {
+          "type": "number",
+          "format": "float",
+          "default": 0.05,
+          "description": "text block detection: minimum area of individual columns (as ratio of total image pixels)"
+        },
+        "columnSepWidthMax": {
+          "type": "number",
+          "format": "float",
+          "default": 0.04,
+          "description": "text block detection: maximum width between individual columns (as ratio of total image width)"
+        },
+        "marginTop": {
+          "type": "number",
+          "format": "float",
+          "default": 0.25,
+          "description": "ruler / edge / text detection: maximum y position to crop from above (as ratio of total image height)"
+        },
+        "marginBottom": {
+          "type": "number",
+          "format": "float",
+          "default": 0.75,
+          "description": "ruler / edge / text detection: minimum y position to crop from below (as ratio of total image height)"
+        },
+        "marginLeft": {
+          "type": "number",
+          "format": "float",
+          "default": 0.3,
+          "description": "ruler / edge / text detection: maximum x position to crop from left (as ratio of total image width)"
+        },
+        "marginRight": {
+          "type": "number",
+          "format": "float",
+          "default": 0.7,
+          "description": "ruler / edge / text detection: minimum x position to crop from right (as ratio of total image width)"
+        },
+        "padding": {
+          "type": "number",
+          "format": "integer",
+          "default": 10,
+          "description": "extend / shrink border resulting from edge detection / text detection by this many px in each direction"
+        }
+      }
+    },
+    "ocrd-anybaseocr-layout-analysis": {
+      "executable": "ocrd-anybaseocr-layout-analysis",
+      "categories": [
+        "Layout analysis"
+      ],
+      "steps": [
+        "layout/analysis"
+      ],
+      "description": "Generates a table-of-content like document structure of the whole document.",
+      "input_file_grp_cardinality": 1,
+      "output_file_grp_cardinality": 0,
+      "resource_locations": ["module"],
+      "parameters": {
+        "batch_size": {
+          "type": "number",
+          "format": "integer",
+          "default": 4,
+          "description": "Batch size for generating test images"
+        },
+        "model_path": {
+          "type": "string",
+          "format": "uri",
+          "content-type": "text/directory",
+          "cacheable": true,
+          "default": "structure_analysis",
```

does not work in the way these are currently packaged (see CI failure)

```suggestion
          "default": "models/structure_analysis",
```

---

Wait. This would work for resolve, but then resmgr list-installed does not show them anymore (because we just disallowed recursion for `module` location in the spec).

What we could instead do in the processor class itself:
```python
    def moduledir(self):
        return resource_filename(self.module, 'models')
```

---

Or just move the models to the package root. I'll try that first, if it works, I'll rebase before merging, so the git repo stays reasonably small.

---

Of course, that would work as well. But redefining moduledir is the pattern we should generally use, I believe. (Also for preinstalled models in ocrd_froc etc.)